tests/misc_env/snmp_destination.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any, Tuple
+
+from paramiko.channel import ChannelFile
+
+from ceph.ceph_admin.helper import GenerateServiceSpec
+from ceph.utils import get_node_by_id
+from utility.log import Log
+
+LOG = Log(__name__)
+
+
+def exec_command(node, **kwargs: Any) -> Tuple:
+    """
+    Return the results of the command executed.
+
+    Args:
+        node:   The node to be used for executing the given command.
+        kwargs: The supported keys are
+                sudo    - Default false, root privilege to be used or not
+                cmd     - The command to be executed
+                check_ec - Evaluate the error code returned.
+
+    Return:
+        out, err    String values
+
+    Raises:
+        CommandFailed   when there is an execution failure
+    """
+    out, err = node.exec_command(**kwargs)
+
+    out = out.read().decode() if isinstance(out, ChannelFile) else out
+    err = err.read().decode() if isinstance(err, ChannelFile) else err
+    return out, err
+
+
+def run(ceph_cluster, **kwargs: Any) -> int:
+    """
+    Test module for setting up snmp destination node.
+
+    Args:
+        ceph_cluster:   The participating Ceph cluster object
+        kwargs:         Supported key value pairs for the key config are
+                        node            | node on which commands to be executed
+                        sudo            | Execute in the privileged mode
+                        check_status    | Default True, if disabled does not raise error
+                        cmd             | Set of commands to be executed
+    Returns:
+        0 - on success
+        1 - on failure
+
+    Raises:
+        CommandError
+
+    """
+    LOG.info("Executing command")
+    config = kwargs["config"]
+    check_status = config.get("check_status", True)
+    sudo = config.get("sudo", False)
+    nodes = config.get("node")
+    node = get_node_by_id(ceph_cluster, nodes)
+    LOG.info(f"node: {node.shortname}")
+    spec_cls = GenerateServiceSpec(
+        node=node, cluster=ceph_cluster, specs=config["specs"]
+    )
+    conf_filename = spec_cls.create_snmp_conf_file()
+    LOG.info(f"conf_file_name:{conf_filename}")
+    conf_copy = f"cp {conf_filename} /root/snmptrapd/snmptrapd_auth.conf"
+    commands = [
```

Comment:
would it make sense to split commands into multiple functions in the same module to accommodate any addintional commands execution in the future(maybe for different SNMP versions).
for example.,
```
def configure_firewalld(): ....
def configure_mib(): ....
... so on
```

Reply:
Updated